components/scream/src/physics/p3/micro_p3_iso_c.f90

```diff
@@ -225,4 +225,35 @@ subroutine access_lookup_table_coll_c(dumjj,dumii,dumj,dumi,index,dum1,dum3,dum4
     call access_lookup_table_coll(dumjj,dumii,dumj,dumi,index,dum1,dum3,dum4,dum5,proc)
   end subroutine access_lookup_table_coll_c
 
+  subroutine cloud_water_conservation_c(qc,qcnuc,dt,qcaut,qcacc,qccol,qcheti,qcshd,     &
+    qiberg,qisub,qidep) bind(C)
+    use micro_p3, only: cloud_water_conservation
+
+    real(kind=c_real), value, intent(in) :: qc, qcnuc, dt
+    real(kind=c_real), intent(inout) :: qcaut, qcacc, qccol, qcheti, qcshd, qiberg, qisub, qidep
+
+    call cloud_water_conservation(qc,qcnuc,dt,qcaut,qcacc,qccol,qcheti,qcshd,qiberg,qisub,qidep)
+  end subroutine cloud_water_conservation_c
+
+  subroutine rain_water_conservation_c(qr,qcaut,qcacc,qimlt,qcshd,dt,    &
+    qrevp,qrcol,qrheti) bind(C)
+    use micro_p3, only: rain_water_conservation
+
+    real(kind=c_real), value, intent(in) :: qr, qcaut, qcacc, qimlt, qcshd, dt
+    real(kind=c_real), intent(inout) :: qrevp, qrcol, qrheti
+
+    call rain_water_conservation(qr,qcaut,qcacc,qimlt,qcshd,dt,qrevp,qrcol,qrheti)
+  end subroutine rain_water_conservation_c
+
+  subroutine ice_water_conservation_c(qitot,qidep,qinuc,qiberg,qrcol,qccol,qrheti,qcheti,dt,    &
+    qisub,qimlt) bind(C)
+    use micro_p3, only: ice_water_conservation
+
+    real(kind=c_real), intent(in) :: qitot, qidep, qinuc, qrcol, qccol, qrheti, qcheti, qiberg, dt
```

Comment:
Inputs should be passed by value if possible.

Reply:
Done.